src/network/message.rs

```diff
@@ -168,15 +168,14 @@ pub struct EditResponseMessage {
 
 #[derive(PartialEq, Clone, Debug, Deserialize, Serialize)]
 pub enum GroupMessage {
-    Add(String),
+    Add(String, usize),
```

Comment:
I had this as an `Option` initially. The reason I changed it is that in the client, I defined the `-p` option is defined like this:
```rust
    Group {
        /// Add a group by name.
        #[clap(short, long, conflicts_with = "remove", value_name = "NAME")]
        add: Option<String>,

        /// Remove a group by name.
        /// This will move all tasks in this group to the default group!
        #[clap(short, long, value_name = "NAME")]
        remove: Option<String>,

        /// The number of tasks the newly added group should run in parallel.
        #[clap(
            short,
            long,
            requires = "add",
            conflicts_with = "remove",
            value_name = "TASKS",
            default_value = "1"
        )]
        parallel_tasks: usize,
    }
```
So it's strictly speaking not optional. It's a bit awkward then to always wrap this in `Some` for the message, but it's obviously possible. The other option, of course, would be to make `parallel_task` an `Option` as well.

Reply:
I'm actually in favor of moving the default to the daemon.
As long as the count is optional in the message as well, the handling shouldn't be too akward, as you can move/clone the option directly into the option.